src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -274,19 +279,32 @@ public void runAppCfgCommand(List<String> args) throws ProcessRunnerException {
   }
 
   /**
-   * Returns the version of the Cloud SDK installation. Unlike other methods in this class that call
-   * gcloud, this method always uses a synchronous ProcessRunner and will block until the gcloud
-   * process returns.
+   * Returns the version of the Cloud SDK installation. Version is determined by reading the VERSION
+   * file located in the Cloud SDK directory.
    *
-   * @throws ProcessRunnerException when process runner encounters an error
+   * @throws CloudSdkOutOfDateException if the Cloud SDK is too out of date to determine its version
+   * @throws IOException if the VERSION file could not be read. This might mean the Cloud SDK
+   *                     installation is corrupted.
    */
-  public CloudSdkVersion getVersion() throws ProcessRunnerException {
-    // gcloud info --format="value(basic.version)"
-    List<String> command = new ImmutableList.Builder<String>().add("info")
-        .addAll(GcloudArgs.get("format", "value(basic.version)")).build();
+  public CloudSdkVersion getVersion() throws IOException {
+    Path versionFile = getSdkPath().resolve(VERSION_FILE);
 
-    String output = runSynchronousGcloudCommand(command);
-    return new CloudSdkVersion(output);
+    if (!Files.isRegularFile(versionFile)) {
+      throw new CloudSdkOutOfDateException(MINIMUM_VERSION);
+    }
+
+    String contents = "";
+    List<String> lines = Files.readAllLines(versionFile, StandardCharsets.UTF_8);
+    if (lines.size() > 0) {
+      // expect only a single line
+      contents = lines.get(0);
+    }
+
+    try {
+      return new CloudSdkVersion(contents);
+    } catch (IllegalArgumentException e) {
+      throw new CloudSdkOutOfDateException(MINIMUM_VERSION, e);
```

Comment:
Just so we're all on the same page, let's discuss the all the possible exception cases so we can use a consistent strategy. Here my new proposal for all cases:

### in CloudSdk.getVersion
1. The version file does not exist:
Throw a CloudSdkOutOfDateException. It's too old to even know what version it is.
2. The version file exists, but we gen an IOException when reading it:
throw IOException
3. The version file exists, but its contents are not parseable:
throw IllegalStateException

### in CloudSdk.validateCloudSdkVersion
1. CloudSdk.getVersion throws a CloudSdkOutOfDateException:
do nothing; callers deal with the CloudSdkOutOfDateException
2. CloudSdk.getVersion throws anything else
wrap in a CloudSdkOutOfDateException. Within the context of specifically only validating the version, perhaps it doesn't make sense to propagate all of the above exception types to the caller, but to just consider this a case where the version is wrong.

@chanseokoh @patflynn What do you think?